integration-cli/docker_api_attach_test.go

```diff
@@ -85,8 +87,10 @@ func (s *DockerSuite) TestGetContainersWsAttachContainerNotFound(c *check.C) {
 	status, body, err := sockRequest("GET", "/containers/doesnotexist/attach/ws", nil)
 	c.Assert(status, checker.Equals, http.StatusNotFound)
 	c.Assert(err, checker.IsNil)
-	expected := "No such container: doesnotexist\n"
-	c.Assert(string(body), checker.Contains, expected)
+	var resp types.ErrorResponse
+	c.Assert(json.Unmarshal(body, &resp), checker.IsNil)
```

Comment:
Agreed. Wasn't sure whether this was a good idea because a lot of other stuff in tests like this is explicit. Also, it might actually be less neat because we probably want a function that gets the message from the error then the assertion to be in the test, which means an extra line to check for errors when unmarshalling. I guess it could also be a go-checker checker.

Aaanyway. I will add a helper in a follow-up PR if I find time. :)